bc_obps/common/tests/endpoints/auth/test_endpoint_permissions.py

```diff
@@ -0,0 +1,281 @@
+            {"method": "get", "endpoint_name": "list_operations"},
+            {"method": "get", "endpoint_name": "list_contacts"},
+            {"method": "get", "endpoint_name": "get_operation", "kwargs": {"operation_id": random_uuid}},
+            {
+                "method": "get",
+                "endpoint_name": "list_facilities_by_operation_id",
+                "kwargs": {"operation_id": random_uuid},
+            },
+            {
+                "method": "get",
+                "endpoint_name": "list_operation_representatives",
+                "kwargs": {"operation_id": random_uuid},
+            },
+            {"method": "get", "endpoint_name": "get_user_operator_by_id", "kwargs": {"user_operator_id": random_uuid}},
+            {"method": "get", "endpoint_name": "get_facility", "kwargs": {"facility_id": random_uuid}},
+            {"method": "get", "endpoint_name": "get_contact", "kwargs": {"contact_id": random_uuid}},
+            {"method": "get", "endpoint_name": "get_registration_purposes"},
+            {
+                "method": "get",
+                "endpoint_name": "get_operation_new_entrant_application",
+                "kwargs": {"operation_id": random_uuid},
+            },
+            {"method": "get", "endpoint_name": "get_operation_v2", "kwargs": {"operation_id": random_uuid}},
+            {"method": "get", "endpoint_name": "get_operation_with_documents", "kwargs": {"operation_id": random_uuid}},
+            {"method": "get", "endpoint_name": "list_operations_v2"},
+            {"method": "get", "endpoint_name": "list_operators_v2"},
+            {
+                "method": "get",
+                "endpoint_name": "get_report_operation_by_version_id",
+                "kwargs": {"version_id": random_int},
+            },
+            {"method": "get", "endpoint_name": "build_form_schema"},
+            {"method": "get", "endpoint_name": "get_dashboard_operations_list"},
+            {
+                "method": "get",
+                "endpoint_name": "get_facility_report_form_data",
+                "kwargs": {"version_id": random_int, "facility_id": random_uuid},
+            },
+            {
+                "method": "get",
+                "endpoint_name": "get_ordered_facility_report_activities",
+                "kwargs": {"version_id": random_int, "facility_id": random_uuid},
+            },
+            {
+                "method": "get",
+                "endpoint_name": "get_facility_report_by_version_id",
+                "kwargs": {"version_id": random_int},
+            },
+        ],
+        "authorized_irc_user_and_industry_admin_user": [
+            {
+                "method": "put",
+                "endpoint_name": "update_user_operator_status",
+                "kwargs": {"user_operator_id": random_uuid},
+            },
+        ],
+    }
+
+    @staticmethod
+    def _call_endpoint(method, endpoint):
+        client_method = getattr(client, method)
```

Comment:
best practice:
- `_call_endpoint` should have a `cls` argument if it's a static class method
- it shouldn't reference an external variable `client` and probably use an internal `cls.client` class attribute instead